core/address.go

```diff
@@ -10,6 +10,10 @@ const numInitCharactersForOnMetachainSC = 5
 
 // IsSmartContractAddress verifies if a set address is of type smart contract
 func IsSmartContractAddress(rcvAddress []byte) bool {
+	if len(rcvAddress) <= hooks.NumInitCharactersForScAddress {
```

Comment:
Can this function be called with something else than a valid address (32 bytes)?